uppsrc/CtrlLib/ScrollBar.cpp

```diff
@@ -73,123 +58,97 @@ ScrollBar::ScrollBar() {
 
 ScrollBar::~ScrollBar() {}
 
-Rect ScrollBar::Slider(int& cc) const
+void ScrollBar::GetSectionInfo(SectionInfo& info)const
 {
+	info.horz = IsHorz();
+	
 	Size sz = GetSize();
-	Rect r;
-	if(IsHorz()) {
-		cc = sz.cx > (3 + style->isleft2 + style->isright2) * style->arrowsize ? style->arrowsize : 0;
-		r = RectC(cc, 0, sz.cx - 2 * cc, sz.cy);
-		if(style->isleft2)
-			r.right -= cc;
-		if(style->isright2)
-			r.left += cc;
-	}
-	else {
-		cc = sz.cy > (3 + style->isup2 + style->isdown2) * style->arrowsize ? style->arrowsize : 0;
-		r = RectC(0, cc, sz.cx, sz.cy - 2 * cc);
-		if(style->isup2)
-			r.bottom -= cc;
-		if(style->isdown2)
-			r.top += cc;
-	}
-	return r;
-}
-
-Rect ScrollBar::Slider() const
-{
-	int dummy;
-	return Slider(dummy);
-}
-
-int& ScrollBar::HV(int& h, int& v) const
-{
-	return IsHorz() ? h : v;
-}
+	int  s0, s1, s2, s3, s5, // width/height of each section (s6==s0)
+		 total, m=3;
```

Comment:
What is the reason of introducing new sections? Can not we just focus at the beginning of getting ride of Buttons that are keep on stack? I mean iterative approach to development. Decrees memory consumption in chunks - not everything at once. 

In this PR we are changing a lot. In my opinion too many things. Merging this might be very risky.

Reply:
As `prev, prev2, next, next2` are removed, `ScrollBar::paint(...), ScrollBar::LeftDown(...), ScrollBar::LeftRepeat`, etc., are required to fake/mock what the individual buttons used to do as containing `Ctrl`s. Before, Slider is divided into 3 parts (eg, in `GetMousePart(...)`), the upper blank erea, thumb, lower blank area. As we now need to paint the space allocated to fake `prev,next, prev2, next2` buttons, 4 more sections are added to create a total of 7 sections.